### PR TITLE
HDX-9868, HDX-9869, HDX-9871 A compendium of small documentation updates, mainly to the docs UI 

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR includes a number of documentation updates relating to various JIRA tickets (listed in title) which impact the docs UI.

1. Overview text is provided for each endpoint - currently this includes a link to a provider website (we may want to include just the name but not the link)

2. Standard descriptive text is provided for {location, admin1, admin2}_ref query parameters - this text needs review

3. The resource endpoint query parameter descriptions are improved in the manner of the dataset endpoint so that the description matches the query parameter terminology rather than the HDX notation.
4. The gender query parameter is described in more detail.
5. population_group and population_status descriptions have been added, although the population_group description is short because it is an enum with many values some of which are cryptic.

### CODE REVIEW

**Review by danmihaila — APPROVED** [review]

**Comment by IanHopkinson** [comment]
@danmihaila I suggest this PR needs approval of at least one of @cjhendrix @davidmegginson @turnerm or @SimonbJohnson as the "documentation team"

File: hdx_hapi/endpoints/get_conflict_events.py [hunk 1]
```diff
@@ -52,7 +52,10 @@ async def get_conflict_events(
     # ref_period_parameters: Annotated[ReferencePeriodParameters, Depends(reference_period_parameters)],
     common_parameters: Annotated[CommonEndpointParams, Depends(common_endpoint_parameters)],
     db: AsyncSession = Depends(get_db),
-    event_type: Annotated[Optional[EventType], Query(description='Event type')] = None,
+    event_type: Annotated[
+        Optional[EventType],
```
Comment (cjhendrix) [inline hdx_hapi/endpoints/get_conflict_events.py:59]: @IanHopkinson : change "on of" to "one of"

File: hdx_hapi/endpoints/get_food_price.py [hunk 2]
```diff
@@ -83,7 +83,10 @@ async def get_food_prices(
     output_format: OutputFormat = OutputFormat.JSON,
 ):
     """
-    List of WFP markets
+    The World Food Programme (WFP) food prices data provides information about food prices for a range of commodities
+    at markets across the world. See the more detailed technical
+    <a href='**http://RTD_SUBCATEGORY_LINK**'>HDX HAPI documentation</a>,
+    and the <a href='https://www.wfp.org/'>original WFP source</a> website.
```
Comment (cjhendrix) [inline hdx_hapi/endpoints/get_food_price.py]: @IanHopkinson please change URL to https://dataviz.vam.wfp.org/economic/prices

File: hdx_hapi/endpoints/get_funding.py [hunk 3]
```diff
@@ -48,6 +48,11 @@ async def get_fundings(
     ] = None,
     output_format: OutputFormat = OutputFormat.JSON,
 ):
+    """
+    UNOCHA's funding data from the Financial Tracking Service providing information on humanitarian aid contributions.
```
Comment (cjhendrix) [inline hdx_hapi/endpoints/get_funding.py]: @IanHopkinson  please change "providing" to "provides"

File: hdx_hapi/endpoints/get_population.py [hunk 4]
```diff
@@ -74,7 +74,9 @@ async def get_populations(
     output_format: OutputFormat = OutputFormat.JSON,
 ):
     """
-    Return the list of populations
+    Population data from the UNOCHA Common Operational Datasets (COD) originally sourced UNFPA (UN Population Fund)
```
Comment (cjhendrix) [inline hdx_hapi/endpoints/get_population.py]: @IanHopkinson Please change to "Baseline population data sourced and maintained by UNFPA (UN Population Fund).

File: hdx_hapi/endpoints/get_population.py [hunk 5]
```diff
@@ -74,7 +74,9 @@ async def get_populations(
...
+    See the more detailed technical <a href='**http://RTD_SUBCATEGORY_LINK**'>HDX HAPI documentation</a>,
+    and the <a href='https://cod.unocha.org/'>UNOCHA COD</a> website.
```
Comment (cjhendrix) [inline hdx_hapi/endpoints/get_population.py]: @IanHopkinson please change URL to "https://data.humdata.org/organization/unfpa"

**Review by cjhendrix — CHANGES_REQUESTED** [review]
@IanHopkinson Comments on several lines in these commits.  Let me know if that's not the most efficient way to do it.

File: hdx_hapi/endpoints/get_conflict_events.py (hunk 1 above)
Comment (IanHopkinson) [reply hdx_hapi/endpoints/get_conflict_events.py:59]: Fixed this

File: hdx_hapi/endpoints/get_food_price.py (hunk 2 above)
Comment (IanHopkinson) [reply hdx_hapi/endpoints/get_food_price.py]: Fixed this

File: hdx_hapi/endpoints/get_population.py (hunk 4 above)
Comment (IanHopkinson) [reply hdx_hapi/endpoints/get_population.py]: Fixed this

**Comment by IanHopkinson** [comment]
@cjhendrix that worked pretty well - I went through my email alerts and fixed them one at a time. I also did the admin2_ref ordering thing which is very simple to fix.

**Comment by cjhendrix** [comment]
I think you can go ahead and merge (at least from a DSys perspective).

**Review by alexandru-m-g — APPROVED** [review]

**Pull request merged**